bamboo-hr.sgnl.yaml

```diff
@@ -0,0 +1,99 @@
+# [Required] ApiVersion: The version of the BambooHR API to use. Currently, only v1 is supported.
+
+# [Required] CompanyDomain: The subdomain used to access your account 
+# (If you access BambooHR at https://mycompany.bamboohr.com, then the subdomain is "mycompany")
+
+# [Optional] OnlyCurrent: If true, only the current employees will be returned.
+# If false, all employees will be returned. This defaults to false.
+
+# [Optional] AttributeMappings: A map of attribute types to potential mappings.
+# The BambooHR SoR Response returns all types as string values. By default, we will attempt to convert 
+# all attributes to the types specified in the catalog. This field allows you to specify additional mappings
+# to perform unconventional type conversions.
+
+# [Optional] AttributeMappings.Date: The format of the date string in the BambooHR response.
+# The BambooHR Admin Console allows you to configure date formats from four options.
+# To find this: Visit the BambooHR Admin Console > Settings > Account > General Settings > Date Input Format (Dropdown)
+# If this field is not specified, we default to "yyyy-mm-dd".
+
+# [Optional] AttributeMappings.Bool: A map of boolean values to their possible string representations.
+# The BambooHR bool type has many possible response values including "0/1", "yes/no", "true/false", etc.
+# This field allows you to specify the possible values for true and false.
+# By default, we will attempt a generic conversion for any attribute that is specified as a bool type in the catalog.
+# Golang recognizes these true/false combinations by default:
+# "1", "t", "T", "true", "TRUE", "True" are recognized as true.
+# "0", "f", "F", "false", "FALSE", "False" are recognized as false.
+
+adapterConfig: ewogICAgImFwaVZlcnNpb24iOiAidjEiLAogICAgImNvbXBhbnlEb21haW4iOiAie3tJbnB1dCBSZXF1aXJlZH19IiwKICAgICJvbmx5Q3VycmVudCI6IGZhbHNlLAogICAgImF0dHJpYnV0ZU1hcHBpbmdzIjogewogICAgICAgICJkYXRlIjogInl5eXktbW0tZGQiLAogICAgICAgICJib29sIjogewogICAgICAgICAgICAidHJ1ZSI6IFsidHJ1ZSJdLAogICAgICAgICAgICAiZmFsc2UiOiBbImZhbHNlIl0KICAgICAgICB9CiAgICB9Cn0=
+auth: # https://documentation.bamboohr.com/docs/getting-started#authentication
+  - basic:
+      username: "{{Input Required: Secret/API key}}"
+      password: "xxx" # Random String
+entities:
+  # https://documentation.bamboohr.com/docs/list-of-field-names
+  # Note: Make sure the externalIDs of the attributes match exactly with the field names in the BambooHR documentation.
+  # Custom fields are prefixed with 'custom' in the SoR - see customBoolField for an example.
+  # The endpoint does not return any errors when fields are misspelled, it simply omits the information.
+  Employee:
+    displayName: Employee
+    externalId: Employee
+    description: Employee Entity in BambooHR
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: id
+        externalId: id
+        type: String
+        indexed: true
+        uniqueId: true
+        attributeAlias: employeeId
```

Comment:
I personally like the attributeAlias since we aren't indexing into an entity's fields when defining the attribute. This relationship is very simple, but the last 3-4 catalog's I've created have used this attributeAlias field so I've been trying to standardize. WDYT?

Reply:
Standardizing SGTM